callbacks.go

```diff
@@ -136,6 +152,105 @@ func (exec *sidecarExecutor) LaunchTask(driver executor.ExecutorDriver, taskInfo
 	go exec.monitorTask(cntnr.ID[:12], taskInfo)
 }
 
+// getMasterHostname talks to the local worker endpoint and discovers the
+// Mesos master hostname.
+func (exec *sidecarExecutor) getMasterHostname() (string, error) {
+	localEndpoint := "http://" + os.Getenv("MESOS_AGENT_ENDPOINT")
+	if len(localEndpoint) < 8 { // Did we get anything in the env var?
+		return "", fmt.Errorf("Can't get MESOS_AGENT_ENDPOINT from env! Won't provide Sidecar seeds.")
+	}
+
+	localStruct := struct {
+		MasterHostname string `json:"master_hostname"`
+	}{}
+
+	// Let's find out the Mesos master's hostname
+	resp, err := exec.fetcher.Get(localEndpoint)
+	defer func() {
+		if resp != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	if err != nil {
```

Comment:
I think this err check should go before the defer above. I think Body can be nil if err != nil.